extensions/interactions/GraphInput/GraphInput.js

```diff
@@ -187,362 +191,468 @@ oppia.directive('graphViz', function() {
+      // permutation is a permutation of [0...permutation.length]
+
+      // Find the pivot and successor
+      var pivot = null;
+      var successor = null;
+      permutation.reduce(function(previousValue, currentValue, currentIndex) {
+        if (previousValue < currentValue) {
+          pivot = currentIndex - 1;
         }
-        if (angular.isDefined(weight) && angular.isNumber(weight)) {
-          $scope.graph.edges[$scope.state.selectedEdge].weight = weight;
+        if (pivot !== null && currentValue > permutation[pivot]) {
+          successor = currentIndex;
         }
-        return $scope.graph.edges[$scope.state.selectedEdge].weight;
-      };
+        return currentValue;
+      });
+      if (pivot === null) return null;
+      var tmp = permutation[pivot];
+      permutation[pivot] = permutation[successor];
+      permutation[successor] = tmp;
+      permutation = permutation.concat(permutation.splice(pivot + 1).reverse());
+      return permutation;
+    };
 
+    // Check against every permutation of vectices.
+    // The new index of vertex i in graph2 is permutation[i].
+    var numVertices = graph2.vertices.length;
+    var permutation = [];
+    for (var i = 0; i < numVertices; i++) {
+      permutation.push(i);
+    }
+    while (permutation !== null) {
+      var foundIsomorphism = true;
+      if (graph1.isLabeled && graph2.vertices.some(function(vertex, index) {
+        return vertex.label != graph1.vertices[permutation[index]].label;
+      })) {
+        foundIsomorphism = false;
+      }
 
-      // Styling functions
-      var DELETE_COLOR = 'red';
-      var HOVER_COLOR = 'aqua';
-      var SELECT_COLOR = 'orange';
-      var DEFAULT_COLOR = 'black';
-      $scope.getEdgeColor = function(index) {
-        if ($scope.state.currentMode === _MODES.DELETE &&
-            index === $scope.state.hoveredEdge &&
-            $scope.canDeleteEdge) {
-          return DELETE_COLOR;
-        } else if (index === $scope.state.hoveredEdge) {
-          return HOVER_COLOR;
-        } else if ($scope.state.selectedEdge === index) {
-          return SELECT_COLOR;
-        } else {
-          return DEFAULT_COLOR;
-        }
-      };
-      $scope.getVertexColor = function(index) {
-        if ($scope.state.currentMode === _MODES.DELETE &&
-            index === $scope.state.hoveredVertex &&
-            $scope.canDeleteVertex) {
-          return DELETE_COLOR;
-        } else if (index === $scope.state.currentlyDraggedVertex) {
-          return HOVER_COLOR;
-        } else if (index === $scope.state.hoveredVertex) {
-          return HOVER_COLOR;
-        } else if ($scope.state.selectedVertex === index) {
-          return SELECT_COLOR;
-        } else {
-          return DEFAULT_COLOR;
+      if (foundIsomorphism) {
+        for (var i = 0; i < numVertices; i++) {
+          for (var j = 0; j < numVertices; j++) {
+            if (adj1[permutation[i]][permutation[j]] != adj2[i][j]) {
```

Comment:
hmm, but comparing 2 edge lists is O(V^2) for each node in the worst case isn't it? (Have to check whether the permuted version of the node exists in the edge list for each node.) But checking adjacency matrix is O(V^2) overall.

Also, the graphs are limited to very small sizes (because of the exponential permutation thing), so I don't think it really matters too much with we use. I think the adjacency matrix one is slightly easier to read too, and I still prefer using that.